### PR TITLE
Support creating templates in a dialog

### PR DESCRIPTION
**Pull request by spadgett**
https://trello.com/c/2A7x5Pbc

Fixes https://github.com/openshift/origin-web-catalog/issues/177
Requires https://github.com/openshift/origin-web-catalog/pull/193

screencast 2017-05-01 13-07-26

Follow-on work:

- [ ] Fix flicker in next steps page. (I don't think this is specific to the dialog.)
- [ ] Preserve entered field values when clicking advanced options
- [x] Don't scroll description with right panel content -- not fixed, but opened https://github.com/openshift/origin-web-catalog/issues/196 to track
- [ ] Support advanced options when creating a new project
- [ ] Update to use angular-patternfly wizard component
- [ ] Improve template dialog styles (I might want your help @rhamilto)

### CODE REVIEW

**Comment by spadgett** [comment]
@ncameronbritt fyi

File: app/scripts/directives/processTemplateDialog.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  };
+
+  $scope.$on('templateInstantiated', function(result) {
+    $scope.selectedProject = result.project;
+    showResults();
+  });
+
+  ctrl.close = function() {
+    var cb = ctrl.onDialogClosed();
+    if (cb) {
```
Comment (jwforres) [inline app/scripts/directives/processTemplateDialog.js]: `if (_.isFunction(cb))`  to be safe?

File: app/views/directives/process-template.html [hunk 2]
```diff
@@ -1,46 +1,28 @@
...
+<fieldset ng-disabled="disableInputs">
+  <form name="$ctrl.templateForm">
+    <template-options is-dialog="$ctrl.isDialog" parameters="$ctrl.template.parameters" expand="true" can-toggle="false">
+      <select-project ng-if="!$ctrl.project" selected-project="$ctrl.selectedProject" name-taken="$ctrl.projectNameTaken"></select-project>
+    </template-options>
+    <!-- TODO: Also support advanced options for new projects. -->
+    <div ng-if="$ctrl.selectedProject.metadata.uid" class="row">
+      <div ng-if="$ctrl.selectedProject.metadata.uid" class="col-sm-8 col-sm-offset-4">
+        <!-- TODO: Preserve entered form values. -->
+        To configure additional template parameters, view
```
Comment (jwforres) [inline app/views/directives/process-template.html]: there are not necessarily additional parameters to configure though.  In some cases labels may be the only thing you get out of going to advanced.

**Comment by spadgett** [comment]
Updated and added dist. @jwforres another look?

**Comment by jwforres** [comment]
[merge]

**Pull request merged**